lib/Connector/XiboDashboardConnector.php

```diff
@@ -483,4 +487,81 @@ public function onWidgetEditOption(WidgetEditOptionRequestEvent $event)
             $event->setOptions($options);
         }
     }
+
+    public function onDataRequest(WidgetDataRequestEvent $event, $eventName, EventDispatcherInterface $dispatcher)
+    {
+        if ($event->getDataProvider()->getDataSource() === 'dashboard') {
+            // Always generate a token
+            $tokenEvent = new XmdsConnectorTokenEvent();
+            $tokenEvent->setTargets($event->getDataProvider()->getDisplayId(), $event->getDataProvider()->getWidgetId());
+            $tokenEvent->setTtl(3600 * 24 * 2);
+            $dispatcher->dispatch($tokenEvent, XmdsConnectorTokenEvent::$NAME);
+            $token = $tokenEvent->getToken();
+
+            if (empty($token)) {
+                throw new ConfigurationException(__('No token returned'));
+            }
+
+            if ($event->getDataProvider()->isPreview()) {
+                $url = $event->getDataProvider()->getRouteParser()->urlFor('layout.preview.connector', [], ['token' => $token]);
```

Comment:
To save having a route parser in here we could output a placeholder `[[connector=$token]]` and parse that out in `WidgetHtmlRenderer::decorateForPreview` ?

I suggest we have a helper for it in the connector trait to output the placeholder in case we change that later? `getLayoutPreviewUrl($token)` ?